src/psfmachine/machine.py

```diff
@@ -681,10 +683,23 @@ def _time_bin(self, npoints=200, downsample=False):
         # within them
         breaks = []
         for spdx in range(len(splits_a)):
-            breaks.append(splits_a[spdx] + np.arange(0, dsplits[spdx]) * npoints)
-        # we include the last cadance as 99% of the sequence lenght
-        breaks.append(int(self.nt * 0.99))
-        breaks = np.hstack(breaks)
+            if not downsample:
+                breaks.append(splits_a[spdx] + np.arange(0, dsplits[spdx]) * npoints)
+            else:
+                # if downsample, the knots are spaced by npoints untill the end of the
+                # segment
+                breaks.append(
+                    np.linspace(
+                        splits_a[spdx],
+                        splits_b[spdx] - int(self.nt * 0.01),
+                        dsplits[spdx],
+                        dtype=int,
+                    )
+                )
+        breaks = np.unique(np.hstack(breaks))
+        # we include the last cadance as 99% of the sequence lenght if necessary
+        if breaks[-1] < self.nt - npoints:
+            breaks = np.append(breaks, int(self.nt * 0.99))
```

Comment:
I'm not sure I understand what this is doing...

Reply:
so it happens that when using the binning option, there's a knot missing at the end of the quarter (this doesn't happen when doing downsampling), so there I'm making sure that if know is missing within the last `npoints`, then I add one at the end. 

Here's an example if I do binning and I comment out lines 701 - 702. 
![image](https://user-images.githubusercontent.com/10449555/158484885-7ff1e183-84ff-4c22-baae-db3ec12110c6.png)

And here an example with downsampling
![image](https://user-images.githubusercontent.com/10449555/158485091-b8f5ec5b-232d-4c5c-a458-962318b93fb2.png)

But maybe is not necessary